.github/workflows/sonarcloud.yml

```diff
@@ -57,13 +51,20 @@ jobs:
           ref: ${{ inputs.head_branch }} # checkout commit that triggered this workflow
           fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of analysis
 
-      # Fetch develop so that Sonar can identify new issues in PR builds
-      - name: "📕 Fetch ${{ inputs.main_branch }}"
-        if: inputs.head_branch != inputs.main_branch
-        run: git rev-parse HEAD && git fetch origin ${{ inputs.main_branch }}:${{ inputs.main_branch }} && git status && git rev-parse HEAD
+      # Fetch base branch from the upstream repo so that Sonar can identify new code in PR builds
+      - name: "📕 Fetch upstream base branch"
+        # workflow_call retains the github context of the caller, so `repository` & `repositoryUrl` will be upstream
+        # always due to it running on `workflow_run` which is called from the context of the target repo and not the fork.
+        if: inputs.base_branch && github.repository != inputs.repo
```

Comment:
If this part of the job only runs on forked PRs, what part of this checks out the base branch on non-forked PRs?